### PR TITLE
compiler: fix non arithmetic distances

### LINKED ISSUE #2163: TypeError: cannot determine truth value of Relational when implementing free surface with Subdomain with simpy==1.12
I got the error below when I tried to implement a free surface boundary condition with the images method roughly following what is done in the acoustic case and tutorial [04_boundary_conditions](https://github.com/devitocodes/devito/blob/master/examples/userapi/04_boundary_conditions.ipynb) . The output shows the sympy version at start (== 1.12). Downgrading to older versions  of sympy as 1.11.1 or 1.10.1 in the devito environment solves the issue. The code to reproduce the issue is in attachment. I will try to write a shorter code (MFE).

```
1.12
Operator `initdamp` ran in 0.01 s
Traceback (most recent call last):
  File "/home/hpc/Documentos/fail_fs_tti.py", line 104, in <module>
    op = Operator(stencils + src_term + rec_term, subs=model.spacing_map)
  File "/home/hpc/my_forks/devito/devito/operator/operator.py", line 158, in __new__
    op = cls._build(expressions, **kwargs)
  File "/home/hpc/my_forks/devito/devito/operator/operator.py", line 180, in _build
    irs, byproduct = cls._lower(expressions, profiler=profiler, **kwargs)
  File "/home/hpc/my_forks/devito/devito/operator/operator.py", line 255, in _lower
    clusters = cls._lower_clusters(expressions, **kwargs)
  File "/home/hpc/my_forks/devito/devito/tools/timing.py", line 76, in __call__
    retval = self.func(*args, **kwargs)
  File "/home/hpc/my_forks/devito/devito/operator/operator.py", line 357, in _lower_clusters
    clusters = cls._specialize_clusters(clusters, **kwargs)
  File "/home/hpc/my_forks/devito/devito/tools/timing.py", line 76, in __call__
    retval = self.func(*args, **kwargs)
  File "/home/hpc/my_forks/devito/devito/core/cpu.py", line 136, in _specialize_clusters
    clusters = cire(clusters, 'invariants', sregistry, options, platform)
  File "/home/hpc/my_forks/devito/devito/tools/timing.py", line 76, in __call__
    retval = self.func(*args, **kwargs)
  File "/home/hpc/my_forks/devito/devito/passes/clusters/aliases.py", line 93, in cire
    clusters = transformer.process(clusters)
  File "/home/hpc/my_forks/devito/devito/passes/clusters/aliases.py", line 235, in process
    return self._process_fatd(clusters, 1, xtracted=[])
  File "/home/hpc/my_forks/devito/devito/ir/clusters/visitors.py", line 111, in _process_fatd
    _clusters = self.callback(list(g), pfx, **kwargs)
  File "/home/hpc/my_forks/devito/devito/passes/clusters/aliases.py", line 256, in callback
    made = self._aliases_from_clusters(g, exclude, ak)
  File "/home/hpc/my_forks/devito/devito/passes/clusters/aliases.py", line 121, in _aliases_from_clusters
    found = collect(mapper.extracted, meta.ispace, self.opt_minstorage)
  File "/home/hpc/my_forks/devito/devito/passes/clusters/aliases.py", line 515, in collect
    found = {g: g.find_rotation_distance(d, interval) for g in impacted}
  File "/home/hpc/my_forks/devito/devito/passes/clusters/aliases.py", line 515, in <dictcomp>
    found = {g: g.find_rotation_distance(d, interval) for g in impacted}
  File "/home/hpc/my_forks/devito/devito/passes/clusters/aliases.py", line 1076, in find_rotation_distance
    min_interval = self._pivot_min_intervals[d].translate(-distance)
  File "/home/hpc/anaconda3/envs/devito_dev/lib/python3.9/site-packages/cached_property.py", line 36, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/home/hpc/my_forks/devito/devito/passes/clusters/aliases.py", line 1172, in _pivot_min_intervals
    ret[d][0] = min(ret[d][0], value)
  File "/home/hpc/anaconda3/envs/devito_dev/lib/python3.9/site-packages/sympy/core/relational.py", line 510, in __bool__
    raise TypeError("cannot determine truth value of Relational")
TypeError: cannot determine truth value of Relational
```
[fail_fs_tti.py.txt](https://github.com/devitocodes/devito/files/12099849/fail_fs_tti.py.txt)

### PR DESCRIPTION
**Pull request by mloubout**
Fix #2163

### CODE REVIEW

File: devito/passes/clusters/aliases.py [hunk 1]
```diff
@@ -1169,8 +1169,12 @@ def _pivot_min_intervals(self):
 
             for d, v in distance:
                 value = v.pop()
-                ret[d][0] = min(ret[d][0], value)
-                ret[d][1] = max(ret[d][1], value)
+                try:
+                    ret[d][0] = min(ret[d][0], value)
+                    ret[d][1] = max(ret[d][1], value)
+                except TypeError:
```
Comment (FabioLuporini) [inline devito/passes/clusters/aliases.py:1175]: Interesting, why can we now end up here with SymPy?

Comment (FabioLuporini) [reply devito/passes/clusters/aliases.py:1175]: I mean SymPy 1.12

Comment (mloubout) [reply devito/passes/clusters/aliases.py:1175]: Somehow previous versions end up with `{0, expr}` as values but with sympy 1.12 you only get `{expr}`. Didn't look too deep into why though

Comment (FabioLuporini) [reply devito/passes/clusters/aliases.py:1175]: there might be a deeper issue then; are we now generating the same exact code for the MFE?

Comment (mloubout) [reply devito/passes/clusters/aliases.py:1175]: We are we test all sympy version 1.9-1.12 that pass the same test

**Comment by FabioLuporini** [comment]
Thanks, merged

**Pull request merged**